packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/events-queue.ts

```diff
@@ -0,0 +1,184 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as sns from '@aws-cdk/aws-sns';
+import * as subscription from '@aws-cdk/aws-sns-subscriptions';
+import * as sqs from '@aws-cdk/aws-sqs';
+import * as cdk from '@aws-cdk/core';
+import { Service } from '../service';
+import { Container } from './container';
+import { ContainerMutatingHook, ServiceExtension } from './extension-interfaces';
+
+// Keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * The settings for the EventsQueue extension.
+ */
+export interface EventsQueueProps {
+  /**
+   * The list of topic subscriptions for this service.
+   *
+   * @default none
+   */
+  readonly topicSubscriptions?: TopicSubscriptionProps[];
+
+  /**
+   * The user-provided default queue for this service.
+   * If the `eventsQueue` is not provided, a default SQS Queue is created for the service.
+   *
+   * @default none
+   */
+  readonly eventsQueue?: sqs.IQueue;
+}
+
+/**
+ * The topic-specific settings for creating the subscription queues.
+ */
+export interface TopicSubscriptionProps {
+  /**
+   * The SNS Topic to subscribe to.
+   */
+  readonly topic: sns.ITopic;
+
+  /**
+   * The user-provided queue to subscribe to the given topic.
+   * If the `queue` is not provided, the default `eventsQueue` will subscribe to the given topic.
+   *
+   * @default none
+   */
+  readonly queue?: sqs.IQueue;
+}
+
+/**
+ * Settings for the hook which mutates the application container
+ * to add the subscription queue URLs to its environment.
+ */
+export interface ContainerMutatingProps {
+  /**
+   * The queue name and URLs to be added to the container environment.
+   */
+  readonly environment: { [key: string]: string };
+}
+
+/**
+ * This hook modifies the application container's environment to
+ * add the queue URLs for the subscription queues of the service.
+ */
+export class SubscribeMutatingHook extends ContainerMutatingHook {
```

Comment:
does `SubscribeMutatingHook` and `ContainerMutatingProps` need to be exported? or is it strictly for internal use